packages/tao/src/commands/ngcli-adapter.ts

```diff
@@ -545,18 +598,60 @@ export class NxScopeHostUsedForWrappedSchematics extends NxScopedHost {
 
   read(path: Path): Observable<FileBuffer> {
     if (isWorkspaceConfigPath(path)) {
-      const match = findWorkspaceConfigFileChange(this.host);
       const nxJsonChange = findMatchingFileChange(this.host, 'nx.json' as Path);
+      const match = findWorkspaceConfigFileChange(this.host);
+
+      let workspaceJsonOverride: Observable<RawWorkspaceJsonConfiguration>;
+      if (
+        this.host.exists('workspace.json') ||
+        this.host.exists('angular.json')
+      ) {
+        if (match) {
+          workspaceJsonOverride = of(parseJson(match.content.toString()));
+        }
+      } else if (!this.__nxInMemoryWorkspace) {
+        // if we've already dealt with this, let NxScopedHost read the cache...
+        // projects created inside a generator will not be visible
+        // to glob when it runs in @nrwl/tao/shared/workspace, so
+        // we have to add them into the file.
+        const createdProjects = findCreatedProjects(this.host);
+        const nxJsonInTree = nxJsonChange
+          ? parseJson(nxJsonChange.content.toString())
+          : parseJson(this.host.read('nx.json').toString());
+        const workspace = buildWorkspaceConfigurationFromGlobs(
```

Comment:
Call 2 times instead of spread operator, same checks as new project-configuration logic